pandas/tests/io/excel/test_writers.py

```diff
@@ -25,6 +25,14 @@
 )
 
 
+@pytest.fixture
+def path(ext):
+    """Fixture to open file for use in each test case.
+    """
+    with ensure_clean(ext) as file_path:
+        yield file_path
+
+
```

Comment:
as a follow on, we can remove this fixture and just the the pytest tmp_path builtin fixture. see #26984

Reply:
How hard would it be to do that here? May be a personal preference but I don't find this way of injecting very clear

Reply:
we're not changing behaviour at the moment by moving the fixture out of the class and splitting.

using the pytest builtin _may_ change behaviour.

```
tmp_path()
Return a temporary directory path object which is unique to each test function invocation, created as a sub directory of the base temporary directory. The returned object is a pathlib.Path object.

Note
in python < 3.6 this is a pathlib2.Path
```
also the filepath won't include a predefined extension like ensure_clean(ext).

This PR already addresses two (sub) issues, so I think probably best to leave this as a follow-on.

Reply:
I guess my main concern with this PR is we are IMO sacrificing readability. I find it harder to reason about what is going on here, especially since this introduces a global fixture that implicitly gets injected by a decorator applied to a test class, which also injects a autouse fixture within the class :-X

Why not just keep all of the setup / teardown logic in one fixture and have the class reference that instead?

Reply:
> Why not just keep all of the setup / teardown logic in one fixture and have the class reference that instead?

that's not pytest idiom. "pytest fixtures: explicit, modular, scalable" https://docs.pytest.org/en/latest/fixture.html

"fixtures are implemented in a modular manner, as each fixture name triggers a fixture function which can itself use other fixtures."

for some reason in pandas we are not using composable fixtures much and hence missing a (maybe many) trick.

also according to the contributing guide we are moving away from classes, "Going forward, we are moving to a more functional style using the pytest framework, which offers a richer testing framework that will facilitate testing and developing. Thus, instead of writing test classes, we will write test functions like this:..." https://dev.pandas.io/docs/development/contributing.html#transitioning-to-pytest

but those two points aside.

The parameterisation of engine and ext for the reader and writer classes is inconsistent and ideally we could share fixtures between the reader and writer tests.

This can't be done with a fixture buried inside a class and not being reusable.

Reply:
> for some reason in pandas we are not using composable fixtures much and hence missing a (maybe many) trick.

we moved to pytest only about 2 years ago, but with a huge setup/teardown infrastructure. We are just now getting around to using fixtures propertly. certainly take logical, modular fixtures. personally love pytest, but *sometimes* it does make things less readable, so we tend to use less complex fixtures at times.

Reply:
so I am currenlty -1 on using pytest tmp_path fixtures, we don't use them *at all*, rather use ``ensure_clean``. I agree we could move to use them, but this is way out of scope here and should be done independently.

Reply:
so marking this as resolved.